File: src/main/java/com/hoolix/processor/models/XYZBasicEvent.java
// package com.hoolix.processor.models;
//
// import com.fasterxml.jackson.annotation.JsonProperty;
// import lombok.AllArgsConstructor;
// import lombok.Getter;
// import org.elasticsearch.action.index.IndexRequest;
// import java.io.Serializable;
// import java.util.HashMap;
// import java.util.List;
// import java.util.Map;
//
// /**
//  * Created by peiyuchao on 2017/1/4.
//  */
// @AllArgsConstructor
// @Getter
// public class XYZBasicEvent extends Event implements Serializable {
//     private final String token;
//     private final String type;
//     private final List<String> tags;
//     private final String message;
//     @JsonProperty("upload_type")
//     private final String uploadType; // "streaming" or "file"
//     @JsonProperty("upload_timestamp")
//     private final Long uploadTimestamp;
//
//     @Override
//     public String getIndexName() {
//         return getToken() + Event.INDEX_NAME_SEPARATOR + getType();
//     }
//
//     @Override
//     public IndexRequest toIndexRequest() {
//         return new IndexRequest(getIndexName(), getType());
//     }
//
//     @Override
//     public Map<String, Object> toPayload() {
//         Map<String, Object> payload = new HashMap<>();
//         payload.put("token", token);
//         payload.put("type", type);
//         payload.put("tags", tags);
//         payload.put("message", message);
//         payload.put("upload_type", uploadType);
//         payload.put("upload_timestamp", uploadTimestamp);
//         return payload;
//     }
// }

File: src/main/scala/com/hoolix/processor/models/XYZBasicEvent.scala
package com.hoolix.processor.models

import org.elasticsearch.action.index.IndexRequest

import scala.collection.JavaConversions


/**
  * Created by peiyuchao on 2017/1/6.
  */
case class XYZBasicEvent(
  token: String,
  `type`: String,
  tags: Seq[String],
  message: String,
  uploadType: String,
  uploadTimestamp: Long // when it is uploaded
) extends Event {

  override def toPayload = collection.mutable.Map(
    "token" -> token,
    "type" -> `type`,
    "tags" -> JavaConversions.asJavaCollection(tags),
    "message" -> message,
    "upload_type" -> uploadType,
    "upload_timestamp" -> uploadTimestamp
  )

  override def indexName = token + "." + `type`
//  getToken + Event.INDEX_NAME_SEPARATOR + getType
  override def indexType = `type`
  override def docId = ???
  override def toIndexRequest = new IndexRequest(indexName, indexType) // let ES create a UUID
}
